universal-nft/docs/protocol.md

```diff
@@ -0,0 +1,93 @@
+# Universal-NFT Protocol Specification
+
+## 1. Overview
+
+Universal-NFT is an extension of the Universal / OPT protocol that defines how non-fungible assets (NFTs) are encoded, identified, and tracked on the Bitcoin blockchain.
+
+The core idea is simple:
+
+- NFTs are represented as a UTF-8 JSON payload inside an OP_RETURN output.
+- The structure mirrors the style of Universal (BRC-20), changing only the protocol identifier and the semantics of the fields.
+
+---
+
+## 2. Payload Format
+
+Every Universal-NFT action is encoded as a JSON object stored in OP_RETURN.
+
+Minimal structure:
+
+{
+  "p":   "nft",
+  "op":  "mint" or "transfer",
+  "tick":"<ticker>",
+  "meta":"<3-hex-tag>",
+  "sig": "<3-hex-tag>"
+}
+
+---
+
+### 2.1 Field Definitions
+
+- **p**  
+  Protocol identifier. Always `"nft"` for this extension.
+
+- **op**  
+  Operation type:  
+  - `"mint"` – creates a new NFT  
+  - `"transfer"` – transfers ownership of an existing NFT  
+
+- **tick**  
+  NFT collection or identifier.  
+  Lowercase alphanumeric plus underscore (`[a-z0-9_]`).  
+  Recommended length: up to 7 characters.
+
+- **meta**  
+  3-character hex tag derived from the SHA-256 hash of the NFT image.  
+  Defined as: last 3 hex characters of the image hash.
+
+- **sig**  
+  3-character deterministic signature derived from a server-side hash (sig_full).  
+  Defined as: last 3 hex characters of the server hash.
+
+- **amt** (optional)  
+  String quantity. Defaults to `"1"`.  
+  For NFTs this value is typically constant and equal to `"1"`.
+
+---
+
+## 3. JSON Encoding Rules
+
+- UTF-8 encoding only.  
+- Must be valid JSON.  
+- No trailing commas.  
+- Field order is not enforced.  
+- Extra fields should be ignored by indexers for forward compatibility.
+
+---
+
+## 4. Validation Rules
+
+A Universal-NFT payload is considered valid if:
+
+1. "p" must be exactly "nft".
+2. "op" must be either "mint" or "transfer".
+3. "tick" must match ^[A-z 0-9]{1,7}$.
```

Comment:
```suggestion
3. "tick" must match ^[a-z0-9]{1,7}$
```

For what was written above about ticker may imply a regex like [a-z 0-9]{1,7} or if we have big caps as well: A-Za-z. 
Moreover, this regex limit to 7 characters, is it the goal? 

In theory current indexer there is a normalisation of tickers on upper case, as if you write small caps the indexer stores big ones. This kind of logic can avoid duplications and limit mint mistakes with caps.